### DISCUSSION TITLE
Skip N number of songs in queue

### DISCUSSION THREAD

**Discussion by solumath** [EvieePy/Wavelink#285] [Q&A] [answered]
I would like to be able to skip a N number of tracks in the queue and move them to history instead of deleting them. Or better jump at the index in queue and move everything behind it to history. Is there a way to do this? From looking into documentation I didn't find a method like that but I might be just blind. 

PS: don't like the idea to call skip in a loop for N times

**Comment by SpookyO** [comment]
> I would like to be able to skip a N number of tracks in the queue and move them to history instead of deleting them. Or better jump at the index in queue and move everything behind it to history. Is there a way to do this? From looking into documentation I didn't find a method like that but I might be just blind.
> 
> PS: don't like the idea to call skip in a loop for N times


Well iirc there is no method like that.
And I think you can just use `Queue.get()` Nth (in a range(n) loop?) time which both removes song from queue and add to history, then play the next song.

**Reply by solumath** [reply]
that would work well but it will call the
```py
 async def on_wavelink_track_start(self, payload: wavelink.TrackStartEventPayload) -> None:
    ...
```
that would not be the best solution seeing multiple messages just being sent into channel

**Reply by SpookyO** [reply, marked as answer]
Well, track_start event ONLY triggers when bot plays something with `Player.play` and you do it for the first time and while autoplay is enabled, library does it for you which triggers track_start even though you are not calling `play` method. `Queue.get()` Only gets next song from queue and ~~puts in history~~, it has nothing to do with the API. And thus, it doesn't trigger track_start event, give yourself a try!

Update: was bit wrong about putting tracks to history while getting track from queue. You have to put track to history manually by using `Queue.history.put()`
```py
for i in range(N):
  t=player.queue.get()
  player.queue.history.put(t)
....
```

**Reply by solumath** [reply]
hmm this sounds that it should work, i will give it a try later and give update, thanks <3

**Reply by solumath** [reply]
works like charm, thanks and sorry for late reply got to it today
